bootloader/bootloader.c

```diff
@@ -139,11 +139,15 @@ void __attribute__((noreturn)) __stack_chk_fail(void)
 
 int main(void)
 {
-	__stack_chk_guard = random32();
 	setup();
 	memory_protect();
 	oledInit();
 
+	// to be extra careful and heed the STM32F205xx Reference manual, Section 20.3.1
+	// we don't use the first random number generated after setting the RNGEN bit in setup
+	random32();
+	__stack_chk_guard = random32(); // this supports compiler provided unpredictable stack protection checks
```

Comment:
I am not sure you can modify `__stack_chk_guard` variable outside of the `main()` function (because the function that sets `__stack_chk_guard` will break the check on return).